src/http/ngx_http_wasm_filter_module.c

```diff
@@ -179,3 +225,163 @@ ngx_http_wasm_body_filter_handler(ngx_http_request_t *r, ngx_chain_t *in)
 
     return rc;
 }
+
+
+static void
+ngx_http_wasm_body_filter_resume(ngx_http_wasm_req_ctx_t *rctx, ngx_chain_t *in)
+{
+    ngx_int_t     rc;
+    ngx_chain_t  *cl;
+
+    rctx->resp_chunk = in;
+    rctx->resp_chunk_len = 0;
+
+    for (cl = rctx->resp_chunk; cl; cl = cl->next) {
+        rctx->resp_chunk_len += ngx_buf_size(cl->buf);
+
+        if (cl->buf->last_buf) {
+            rctx->resp_chunk_eof = 1;
+            break;
+
+        } else if (cl->buf->last_in_chain) {
+            if (rctx->r != rctx->r->main) {
+                rctx->resp_chunk_eof = 1;
+            }
+
+            break;
+        }
+    }
```

Comment:
This loop is the last thing that I would like to improve before merging this, because it may have impact into how we may reuse chained links in the future; this loop attaches buffers that do not belong to ngx_wasm_module, to the response body buffers, which do belong to ngx_wasm_module. Sometimes, modules don't set the buffer's `tag` (e.g. echo-nginx-module), so distinguishing where buffers come from becomes difficult. I'll have to mull over it and see how this can ensure buffer reuse in the future (or not, perhaps).